src/distilabel/steps/tasks/argilla_labeller.py

```diff
@@ -0,0 +1,614 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+import orjson as json
+from argilla import (
+    LabelQuestion,
+    MultiLabelQuestion,
+    RatingQuestion,
+    Record,
+    SpanQuestion,
+    Suggestion,
+    TextField,
+    TextQuestion,
+)
+from argilla._models._settings._questions import (
+    LabelQuestionSettings,
+    MultiLabelQuestionSettings,
+    RatingQuestionSettings,
+    TextQuestionSettings,
+)
+from jinja2 import Template
+from pydantic import BaseModel, Field, PrivateAttr
+from typing_extensions import override
+
+from distilabel.mixins.runtime_parameters import RuntimeParameter
+from distilabel.steps.base import StepInput
+from distilabel.steps.tasks.base import Task
+from distilabel.steps.tasks.typing import ChatType
+
+if sys.version_info < (3, 9):
+    import importlib_resources
+else:
+    import importlib.resources as importlib_resources
+
+if TYPE_CHECKING:
+    from distilabel.steps.typing import StepOutput
+
+_LABELQUESTION_SETTINGS = LabelQuestionSettings()
+_MULTILABELQUESTION_SETTINGS = MultiLabelQuestionSettings()
+_TEXTQUESTION_SETTINGS = TextQuestionSettings()
+_RATINGQUESTION_SETTINGS = RatingQuestionSettings(
+    options=[
+        {"value": 1, "name": "1"},
+        {"value": 2, "name": "2"},
+    ]
+)
```

Comment:
We should make all these Argilla imports within the `ArgillaLabeller` class, otherwise this will raise an import error if you don't have installed argilla even when you're not trying to use this task as we're importing it in `distilabel.steps.tasks.__init__`.